repo/db/purchases.go

```diff
@@ -119,17 +120,55 @@ func (p *PurchasesDB) Delete(orderID string) error {
 	return nil
 }
 
-func (p *PurchasesDB) GetAll(offsetId string, limit int) ([]repo.Purchase, error) {
+func (p *PurchasesDB) GetAll(offsetId string, limit int, stateFilter []pb.OrderState, searchTerm string) ([]repo.Purchase, error) {
 	p.lock.RLock()
 	defer p.lock.RUnlock()
 
+	stateFilterClause := ""
```

Comment:
IMO all this state/filter/query building stuff should be in a single function as well.